src/base/QXmppMixItems.cpp

```diff
@@ -1,19 +1,960 @@
+static const auto ADMINISTRATOR_JIDS_KEY = QStringLiteral("Administrator");
+static const auto CHANNEL_DELETION_KEY = QStringLiteral("End of Life");
+static const auto NODES_KEY = QStringLiteral("Nodes Present");
+static const auto MESSAGES_SUBSCRIBE_ROLE_KEY = QStringLiteral("Messages Node Subscription");
+static const auto MESSAGES_RETRACT_ROLE_KEY = QStringLiteral("Administrator Message Retraction Rights");
+static const auto PRESENCE_SUBSCRIBE_ROLE_KEY = QStringLiteral("Presence Node Subscription");
+static const auto PARTICIPANTS_SUBSCRIBE_ROLE_KEY = QStringLiteral("Participants Node Subscription");
+static const auto INFORMATION_SUBSCRIBE_ROLE_KEY = QStringLiteral("Information Node Subscription");
+static const auto INFORMATION_UPDATE_ROLE_KEY = QStringLiteral("Information Node Update Rights");
+static const auto ALLOWED_JIDS_SUBSCRIBE_ROLE_KEY = QStringLiteral("Allowed Node Subscription");
+static const auto BANNED_JIDS_SUBSCRIBE_ROLE_KEY = QStringLiteral("Banned Node Subscription");
+static const auto CONFIGURATION_READ_ROLE_KEY = QStringLiteral("Configuration Node Access");
+static const auto AVATARS_UPDATE_ROLE_KEY = QStringLiteral("Avatar Nodes Update Rights");
+static const auto NICKNAME_REQUIRED_KEY = QStringLiteral("Mandatory Nicks");
+static const auto PRESENCE_REQUIRED_KEY = QStringLiteral("Participants Must Provide Presence");
+static const auto ONLY_PARTICIPANTS_PERMITTED_TO_SUBMIT_PRESENCE_KEY = QStringLiteral("Open Presence");
+static const auto OWN_MESSAGE_RETRACTION_PERMITTED_KEY = QStringLiteral("User Message Retraction");
+static const auto INVITATIONS_PERMITTED_KEY = QStringLiteral("Participation Addition by Invitation from Participant");
+static const auto PRIVATE_MESSAGES_PERMITTED_KEY = QStringLiteral("Private Messages");
+
+static const QMap<QXmppMixConfigItem::Role, QString> ROLES = {
+    { QXmppMixConfigItem::Role::Owner, QStringLiteral("owners") },
+    { QXmppMixConfigItem::Role::Administrator, QStringLiteral("admins") },
+    { QXmppMixConfigItem::Role::Participant, QStringLiteral("participants") },
+    { QXmppMixConfigItem::Role::Allowed, QStringLiteral("allowed") },
+    { QXmppMixConfigItem::Role::Anyone, QStringLiteral("anyone") },
+    { QXmppMixConfigItem::Role::Nobody, QStringLiteral("nobody") },
+};
+
+static const QMap<QXmppMixConfigItem::Node, QString> NODES = {
+    { QXmppMixConfigItem::Node::AllowedJids, QStringLiteral("allowed") },
+    { QXmppMixConfigItem::Node::AvatarData, QStringLiteral("avatar") },
+    { QXmppMixConfigItem::Node::AvatarMetadata, QStringLiteral("avatar") },
+    { QXmppMixConfigItem::Node::BannedJids, QStringLiteral("banned") },
+    { QXmppMixConfigItem::Node::Information, QStringLiteral("information") },
+    { QXmppMixConfigItem::Node::JidMap, QStringLiteral("jidmap-visible") },
+    { QXmppMixConfigItem::Node::Participants, QStringLiteral("participants") },
+    { QXmppMixConfigItem::Node::Presence, QStringLiteral("presence") },
+};
+
```

Comment:
The modern way would be to use std::array with QStringViews and `enumFromString<QXmppMixConfigItem::Node>` as in https://github.com/qxmpp-project/qxmpp/commit/c0f9a28340187eb8bd67f12878998f5e901465d6.

but I guess it's okay to keep it like this for now, QMaps are used in many other cases too.

Reply:
I am in favor of a common mechanisms to map the enumerators to strings. But I think that having two lists instead of a mapping makes it harder to maintain the relations between the enumerators and the strings.